Cpp_hands_on_programming/lib/abstract/Ipc_method.h

```diff
@@ -0,0 +1,20 @@
+#ifndef IPC_METHOD
+#define IPC_METHOD
+
+#include "lib/ipc_info/Ipc_info.h"
+#include <vector>
+#include <fstream>
+#include <thread>
+#include <chrono>
```

Comment:
I personally prefer to add includes into files which are actually needs them, instead of doing it in main header. You may save some microseconds on compile time 🙂

Reply:
Oh never know it could make a difference in compile time, thanks for the tips. Does it mean for example if I have Ipc_info.h and Ipc_info.cc, I should include directly the necessary file for Ipc_info.cc in Ipc_info.cc instead of include it in Ipc_info.h and get the header indirectly from `#include` "Ipc_info.h`?

Reply:
Yes. Of course improvement in compile time would be usually not noticeable, as compilers likely cache headers and do other optimizations. But it's also good for other reasons. Your code will be clearer and easier to maintain, especially in large projects. I wanted to add some example, but it's easier to just post SO 😛 https://stackoverflow.com/a/2596554